executor/point_get.go

```diff
@@ -313,19 +320,19 @@ func encodeIndexKey(e *baseExecutor, tblInfo *model.TableInfo, idxInfo *model.In
 		} else {
 			idxVals[i], err = table.CastValue(e.ctx, idxVals[i], colInfo, true, false)
 			if types.ErrOverflow.Equal(err) {
-				return nil, kv.ErrNotExist
+				return nil, false, kv.ErrNotExist
```

Comment:
We could return `nil, true, nil`, then outer do not need to check `kv.ErrNotExist`.

Reply:
After discussing with @tiancaiamao , I change the argument `nonExist` to `hasNull`. PTAL